app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,31 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = @current_user.posts.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(post_params)
```

Comment:
- [ ] It will be great to create a post on behalf of the currently logged-in user. At the moment in this assignment, we are assuming the currently logged-in user is the one that is defined in your application controller as the `current_user` method. So when you are creating your post, don't get the user_id from your params get the user id from the `@current_user`. Hint:  
`@post = @current_user.posts.new(post_params)`